crates/ruff/src/rules/refurb/rules/repeated_append.rs

```diff
@@ -0,0 +1,377 @@
+use std::collections::HashMap;
+
+use ast::{traversal, ParameterWithDefault, Parameters, Ranged};
+use ruff_diagnostics::{AutofixKind, Diagnostic, Edit, Fix, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{self as ast, Expr, ExprName, Stmt};
+use ruff_python_codegen::Generator;
+use ruff_python_semantic::analyze::type_inference::{PythonType, ResolvedPythonType};
+use ruff_python_semantic::{Binding, BindingKind, Definition, DefinitionId, NodeId};
+use ruff_text_size::TextRange;
+
+use crate::checkers::ast::Checker;
+use crate::registry::AsRule;
+
+/// ## What it does
+/// Checks for consecutive calls to `append`.
+///
+/// ## Why is this bad?
+/// Consecutive calls to `append` can be less efficient than using `extend` because each
+/// `append` re-sizes the list individually, whereas `extend` re-sizes it once for all elements.
+///
+/// ## Example
+/// ```python
+/// nums = [1, 2, 3]
+///
+/// nums.append(4)
+/// nums.append(5)
+/// nums.append(6)
+/// ```
+///
+/// Use instead:
+/// ```python
+/// nums = [1, 2, 3]
+///
+/// nums.extend((4, 5, 6))
+/// ```
+///
+/// ## References
+/// - [Python documentation: More on Lists](https://docs.python.org/3/tutorial/datastructures.html#more-on-lists)
+#[violation]
+pub struct RepeatedAppend {
+    name: String,
+    replacement: String,
+}
+
+impl Violation for RepeatedAppend {
+    const AUTOFIX: AutofixKind = AutofixKind::Sometimes;
+
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let RepeatedAppend { name, replacement } = self;
+        format!("Use `{replacement}` instead of repeatedly calling `{name}.append()`")
+    }
+
+    fn autofix_title(&self) -> Option<String> {
+        let RepeatedAppend { replacement, .. } = self;
+        Some(format!("Replace with `{replacement}`"))
+    }
+}
+
+pub(crate) fn convert_consecutive_appends_into_extend(checker: &mut Checker, stmt: &Stmt) {
```

Comment:
Nit: convention is to name these identically to the rule (like `repeated_append`).

Reply:
I even thought about it when implementing next checkers 😅 will do!